packages/service-library/src/servicelib/redis_utils.py

```diff
@@ -1,25 +1,34 @@
-# FIXME: move to settings-library or refactor
-
+import functools
 import logging
-from typing import Optional
+from typing import Optional, Union
 
-from tenacity import before_sleep_log, stop_after_attempt, wait_fixed
+from .redis import RedisClientSDK
 
 log = logging.getLogger(__file__)
 
 
-class RedisRetryPolicyUponInitialization:
-    """Retry policy upon service initialization"""
+def exclusive(
+    redis: RedisClientSDK,
+    *,
+    lock_key: str,
+    lock_value: Optional[Union[bytes, str]] = None
+):
+    """
+    Define a method to run exclusively accross
+    processes by leveraging a Redis Lock.
+
+    parameters:
+    redis: the redis client SDK
+    lock_key: a string as the name of the lock (good practice: app_name:lock_name)
+    lock_value: some additional data that can be retrieved by another client
+    """
 
-    WAIT_SECS = 2
-    ATTEMPTS_COUNT = 20
+    def decorator(func):
+        @functools.wraps(func)
+        async def wrapper(*args, **kwargs):
+            async with redis.lock_context(lock_key=lock_key, lock_value=lock_value):
```

Comment:
Q: so this ``lock_context``ensures as well unlocking if the service is restarted, right?

Reply:
Q: if redis has a lock, does it wait or raises?

Reply:
yes, the lock is TTLed

Reply:
> Q: if redis has a lock, does it wait or raises?

Could you precise this? what do you mean by redis has a lock?

Reply:
I think he meant if the lock in redid is already locked, what happens?

Reply:
ok, it will raise ```AlreadyLockedError```